src/openrct2-ui/windows/TitleCommandEditor.cpp

```diff
@@ -613,62 +613,55 @@ static void window_title_command_editor_tool_down(
 
     if (info.type == VIEWPORT_INTERACTION_ITEM_SPRITE)
     {
-        uint16_t spriteIndex = info.sprite->generic.sprite_index;
-        uint16_t spriteIdentifier = info.sprite->generic.sprite_identifier;
         bool validSprite = false;
-        if (info.sprite->generic.Is<Peep>())
+        auto peep = info.sprite->As<Peep>();
+        auto vehicle = info.sprite->As<Vehicle>();
+        auto litter = info.sprite->As<Litter>();
+        auto duck = info.sprite->As<Duck>();
+        auto balloon = info.sprite->As<Balloon>();
+        if (peep != nullptr)
```

Comment:
Can't this become a `switch()...case` instead?

Reply:
I was getting fed up with all the extra nullptr checks involved with doing a switch.

Reply:
You can also do something like ```if(auto peep = info.sprite->As<Peep>(); peep != nullptr)``` since C++17.

Reply:
I used that before on another pr and it wasn't liked by a few folk.